src/sfizz/Synth.cpp

```diff
@@ -1036,11 +1005,12 @@ void sfz::Synth::disableFreeWheeling() noexcept
 
 void sfz::Synth::resetAllControllers(int delay) noexcept
 {
-    AtomicGuard callbackGuard { inCallback };
-    if (!canEnterCallback)
+    resources.midiState.resetAllControllers(delay);
```

Comment:
Is this really safe to reset controllers without a lock?
this call will be enqueuing MIDI events, right?

Reply:
I have to admit that this does not feel entirely right. The callback guard is disabled when voices or regions are touched, but I was thinking that even in this case you could still track midi events, such as allnotesoff. The controller reset does not do anything on the midi state but put a zero event on all controllers after a delay.

Reply:
The issue would probably be that the midi time is not advanced in the render method. The try lock failure should still advance the midi time.

Reply:
The validity of this model is questionable maybe.
Also it seems more a problem for a different issue.

It will be a behavior no different of noteOn/Off/etc, and also what's currently of sfizz.
Is that correct?

Reply:
All midi events basically get passed to the midi state now, but if the callback lock cannot be taken they have no impact on the voices, regions, or anything. The current behavior of sfizz should be the same except it used the AtomicGuards.

I agree, this may not be the ideal behavior, but I could not really make up my mind so I went with this solution. The thing is that in practice you would kind of expect that a plugin "knows" the current state of all controllers for example. I am not sure that host resend all controller state upon playback.

Anyway, if you find even one reason to change it we can. Most cases where the lock is taken would be so devastating that we'd be better off clearing the midi state in most cases anyway.